components/Discussion/DiscussionPreferences.js

```diff
@@ -59,151 +64,301 @@ DiscussionPreferences.propTypes = {
+                  }}
+                >
+                  {t('components/DiscussionPreferences/commentAnonymously')}
+                </Checkbox>
+              </div>
             </div>
           )}
-          <Button onClick={this.onSave}>
+          <div {...styles.fieldWrapper}>
+            <Interaction.H2>
+              {t('components/DiscussionPreferences/credentialHeading')}
+            </Interaction.H2>
+            <Label>
+              {t('components/DiscussionPreferences/credential/description')}
+            </Label>
+            <Field
+              label={t('components/DiscussionPreferences/credentialLabel')}
+              value={state.credential}
+              onChange={(_, val) => {
+                setState(curr => ({
+                  ...curr,
+                  credential: val
+                }))
+              }}
+            />
+            {isListedCredential && state.anonymity && (
+              <div style={{ marginBottom: 10 }}>
+                <Label>
+                  {t(
+                    'components/DiscussionPreferences/credentialAnonymityWarning'
+                  )}
+                </Label>
+              </div>
+            )}
+            {credentialSuggestions && (
+              <div {...styles.suggestedCredentialsWrapper}>
+                <Interaction.H3>
+                  {t(
+                    'components/DiscussionPreferences/existingCredentialLabel'
+                  )}
+                </Interaction.H3>
+                <div {...styles.suggestedCredentialsContainer}>
+                  {credentialSuggestions
+                    .slice(
+                      0,
+                      showAllSuggestedCredentials
+                        ? credentialSuggestions.length - 1
+                        : 4
+                    )
+                    .map(item => (
+                      <button
+                        key={item.description}
+                        onClick={() =>
+                          setState(curr => ({
+                            ...curr,
+                            credential: item.description
+                          }))
+                        }
+                        {...suggestedCredentialButtonStyling}
+                      >
+                        <Credential
+                          {...item}
+                          textColor={colorScheme.getCSSColor('text')}
+                        />
+                      </button>
+                    ))}
+                </div>
+                {!showAllSuggestedCredentials &&
+                  credentialSuggestions.length >= 5 && (
+                    <A
+                      href='#'
+                      onClick={() => setShowAllSuggestedCredentials(true)}
+                    >
+                      {t('components/DiscussionPreferences/showAllCredentials')}
+                    </A>
+                  )}
+              </div>
+            )}
+          </div>
+          <Button type='submit'>
```

Comment:
Adding the `block`prop to the Button could look nice on mobile (optional)
<img width="321" alt="Screenshot 2021-11-26 at 15 10 15" src="https://user-images.githubusercontent.com/20746301/143593360-2ee16283-17ea-40eb-a920-c63b389e3276.png">